week-four/app/src/main/java/pro/marvinhosea/movielist/ui/movies/MovieListFragment.kt

```diff
@@ -0,0 +1,164 @@
+import pro.marvinhosea.movielist.R
+import pro.marvinhosea.movielist.ui.movies.show.MovieDetailActivity
+import pro.marvinhosea.movielist.adapters.MovieAdapter
+import pro.marvinhosea.movielist.data.models.Movie
+import pro.marvinhosea.movielist.data.models.Success
+import pro.marvinhosea.movielist.data.models.response.Result
+import pro.marvinhosea.movielist.networking.NetworkStatusChecker
+import pro.marvinhosea.movielist.networking.RemoteApi
+import pro.marvinhosea.movielist.networking.buildApiService
+import pro.marvinhosea.movielist.repository.UserSharedPrefRepository
+import pro.marvinhosea.movielist.ui.login.UserLoginActivity
+import pro.marvinhosea.movielist.utils.toast
+
+class MovieListFragment : Fragment(), MovieAdapter.MovieListClickListener {
+    private val movieViewModel by lazy {
+        ViewModelProvider(this).get(MovieViewModel::class.java)
+    }
+    private var userSharedRepository = UserSharedPrefRepository
+
+    private val adapter by lazy { MovieAdapter(mutableListOf(), this) }
+
+    private val networkStatusChecker by lazy {
+        activity?.getSystemService(ConnectivityManager::class.java)?.let { NetworkStatusChecker(it) }
+    }
+
+    private val apiService by lazy { buildApiService() }
+
+    private val remoteApi by lazy {
+        RemoteApi(apiService)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        movieRecyclerView.adapter = adapter
+        if (!networkStatusChecker?.hasInternetConnection()!!) {
+            requireContext().toast("No Internet Connection")
+            lifecycleScope.launch {
+                adapter.setData(movieViewModel.getAllMovies())
+            }
+            return
+        }
+        getMovies("Top Rated Movies")
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.menu_main, menu)
+        super.onCreateOptionsMenu(menu, inflater)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.watch_list -> myWatchList()
+            R.id.top_rate -> getMovies("Top Rated Movies")
+            R.id.popular_movies -> getMovies("Popular movies")
+            R.id.now_playing -> getMovies("Now Playing Movies")
+            R.id.upcoming_movies -> getMovies("Upcoming Movies")
+            R.id.logout_menu -> logOut()
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    private fun myWatchList() {
+        lifecycleScope.launch {
+            (activity as AppCompatActivity).supportActionBar?.title = "My Watchlist"
+            adapter.setData(movieViewModel.myWatchListMovies())
+        }
+    }
+
+    private fun logOut() {
+        activity?.applicationContext?.let { userSharedRepository.init(it) }
```

Comment:
Not sure why you went this route, if you had provided the preferences/Repo within the App class, and reused it, you wouldn't have to recreate/reinit the repo all the time.